code/game/objects/items/devices/scanners.dm

```diff
@@ -226,20 +226,26 @@ REAGENT SCANNER
 			for(var/datum/wound/W in e.wounds) if(W.internal)
 				user.show_message(text("\red Internal bleeding detected. Advanced scanner required for location."), 1)
 				break
-		if(H.vessel)
-			var/blood_type = H.get_blood_name()
-			var/blood_volume = round(H.vessel.get_reagent_amount(blood_type))
-			var/blood_percent =  blood_volume / BLOOD_VOLUME_NORMAL
-			blood_percent *= 100
-			if(blood_volume <= 500)
-				user.show_message("\red <b>Warning: Blood Level LOW: [blood_percent]% [blood_volume]cl")
-			else if(blood_volume <= 336)
-				user.show_message("\red <b>Warning: Blood Level CRITICAL: [blood_percent]% [blood_volume]cl")
+		var/blood_id = H.get_blood_id()
+		if(blood_id)
+			if(ishuman(H))
+				if(H.bleed_rate)
+					to_chat(user, "<span class='danger'>Subject is bleeding!</span>")
+			var/blood_percent =  round((H.blood_volume / BLOOD_VOLUME_NORMAL)*100)
+			var/blood_type = H.b_type
+			if(blood_id != "blood")//special blood substance
+				blood_type = blood_id
+			if(H.blood_volume <= BLOOD_VOLUME_SAFE && H.blood_volume > BLOOD_VOLUME_OKAY)
+				to_chat(user, "<span class='danger'>LOW blood level [blood_percent] %, [H.blood_volume] cl,</span> <span class='info'>type: [blood_type]</span>")

```

Comment:
This should probably use user.show_message, as should the rest, just for consistency